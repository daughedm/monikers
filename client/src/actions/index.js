import {
  getTeamNames,
  numOfPlayers,
  currentTeam,
  currentRound,
  teamOneScore,
  teamTwoScore,
  updateTeamTimer,
  addTeamNames,
  resetStore
} from './gameActions/gameActions';

import {
  updateActiveCards,
  discardedCards,
  addCard,
  numOfCards,
  clearDiscardedCards
} from './cardActions/cardActions';

export {
  getTeamNames,
  numOfPlayers,
  addCard,
  discardedCards,
  clearDiscardedCards,
  currentTeam,
  currentRound,
  teamOneScore,
  teamTwoScore,
  updateActiveCards,
  updateTeamTimer,
  numOfCards,
  addTeamNames,
  resetStore
};
