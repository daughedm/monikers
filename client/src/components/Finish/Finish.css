.animation-two {
  background-image: url('../../assets/monikers_furry_one-01.svg');
  background-repeat: no-repeat;
  background-size: 100%;
  height: 600px;
  position: absolute;
  bottom: -600px;
  width: 100%;
  animation: person 10s infinite;
}

.animation-one {
  background-image: url('../../assets/furry-one-noBG-02.svg');
  background-repeat: no-repeat;
  background-size: 100%;
  height: 600px;
  width: 100%;
  position: absolute;
  bottom: -600px;
  animation: furry 10s infinite;
}

.win-headline {
  text-align: center;
  font-size: 4rem;
  margin: 0;
}

.win-headline-two {
  text-align: center;
  font-size: 2.5rem;
  margin: 0;
  color: #555;
}

.start-newgame-button {
  width: 100%;
  height: 40px;
  color: #FFF;
  background-color: rgba(76, 189, 159, 1);
  border: 1px solid #FFF;
  border-radius: 5px;
  font-size: 1rem;
  margin-top: 0;
  -webkit-box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
  -moz-box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
  box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
}

.finish-transition {
  width: 85%;
  margin: 40px auto 25px auto;
  background-color: #FFF;
  background-size: 75%;
  border-radius: 15px;
  padding: 40px 20px 40px 20px;
  -webkit-box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
  -moz-box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
  box-shadow: 0px 2px 12px 0px rgba(0,0,0,0.37);
}

@keyframes furry {
  0%   {
    bottom: -600px;
  }
  25%  {
    bottom: -350px;
  }

  50%  {
    bottom: -600px;
  }
  100% {
    bottom: -600px;
  }
}

@keyframes person {
  0%   {
    bottom: -600px;
  }
 

  50%  {
    bottom: -600px;
  }
  75%  {
    bottom: -350px;
  }
  100% {
    bottom: -600px;
  }
}

@media (min-width: 410px) {
  .finish-transition {
    width: 80%;
    padding: 40px 30px 40px 30px;
  }

  .start-newgame-button {
    height: 45px;
  }
}