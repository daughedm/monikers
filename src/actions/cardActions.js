export const numOfCards = number => ({
  type: 'NUM_OF_CARDS',
  number
});

export const addCard = card => ({
  type: 'ADD_CARD',
  card
});

export const updateActiveCards = cards => ({
  type: 'UPDATE_ACTIVE_CARDS',
  cards
})

export const discardedCards = card => ({
  type: 'DISCARDED_CARDS',
  card
});

export const clearDiscardedCards = card => ({
  type: 'CLEAR_DISCARDED_CARDS',
  card
});
