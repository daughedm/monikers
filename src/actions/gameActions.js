export const getTeamNames = teamName => ({
  type: 'TEAM_NAMES',
  teamName
});

export const numOfPlayers = number => ({
  type: 'NUM_OF_PLAYERS',
  number
});

export const addCard = card => ({
  type: 'ADD_CARD',
  card
});

export const discardedCards = card => ({
  type: 'DISCARDED_CARDS',
  card
});

export const currentTeam = currentTeam => ({
  type: 'CURRENT_TEAM',
  currentTeam
});

export const currentRound = roundNumber => ({
  type: 'CURRENT_ROUND',
  roundNumber
});

export const teamOneScore = points => ({
  type: 'TEAM_ONE_SCORE',
  points
});

export const teamTwoScore = points => ({
  type: 'TEAM_TWO_SCORE',
  points
});
