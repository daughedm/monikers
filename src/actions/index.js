import {
  getTeamNames,
  numOfPlayers
} from './gameActions.js';

export  {
  getTeamNames,
  numOfPlayers
};
