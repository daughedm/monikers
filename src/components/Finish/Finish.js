import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import * as actions from '../../actions';

export class Finish extends Component {
  handleSubmit = e => {
    e.preventDefault();
    this.props.resetStore();
    this.props.history.push('/setup');
  };

  render() {
    const { teamNames, teamOneScore, teamTwoScore } = this.props;
    const winningTeam = teamOneScore > teamTwoScore ? 0 : 1;
    return (
      <div className="background-monikers">
        <div className="round-transition">
          <h2 className="round-headline">
            Team {teamNames[winningTeam]} wins!
          </h2>
          <div className="dashed-line" />
          <button
            className="start-button"
            type="submit"
            onClick={this.handleSubmit}
          >
            NEW GAME
          </button>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  teamNames: state.teamNames,
  teamOneScore: state.teamOneScore,
  teamTwoScore: state.teamTwoScore
});

export const mapDispatchToProps = dispatch => ({
  resetStore: () => dispatch(actions.resetStore())
});

Finish.propTypes = {
  teamNames: PropTypes.array,
  teamOneScore: PropTypes.number,
  teamTwoScore: PropTypes.number,
  resetStore: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Finish)
);
