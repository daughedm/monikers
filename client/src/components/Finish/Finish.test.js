/* eslint-disable max-len */

import React from 'react';
import { shallow } from 'enzyme';
import { Finish, mapStateToProps, mapDispatchToProps } from './Finish';
import * as actions from '../../actions/';
describe('finish', () => {
  let finish;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      history: { push: jest.fn() },
      teamNames: ['team1', 'team2'],
      teamOneScore: 1,
      teamTwoScore: 2,
      resetStore: jest.fn()
    };
    finish = shallow(<Finish {...mockProps} />);
  });

  it('matches the snapshot', () => {
    expect(finish).toMatchSnapshot();
  });

  it('matches the snapshot if teamOneScore is higher', () => {
    mockProps.teamOneScore = 3;
    finish = shallow(<Finish {...mockProps} />);

    expect(finish).toMatchSnapshot();
  });

  describe('handleSubmit', () => {
    it('calls preventDefault', () => {
      const mockEvent = { preventDefault: jest.fn() };

      finish.instance().handleSubmit(mockEvent);

      expect(mockEvent.preventDefault).toHaveBeenCalledTimes(1);
    });

    it('calls resetStore', () => {
      const mockEvent = { preventDefault: jest.fn() };

      finish.instance().handleSubmit(mockEvent);

      expect(mockProps.resetStore).toHaveBeenCalledTimes(1);
    });

    it('calls history.push', () => {
      const mockEvent = { preventDefault: jest.fn() };

      finish.instance().handleSubmit(mockEvent);

      expect(mockProps.history.push).toHaveBeenCalledTimes(1);
    });
  });

  describe('mapStateToProps', () => {
    it('maps state properties to props', () => {
      const mockState = {
        teamNames: actions.teamNames,
        teamOneScore: actions.teamOneScore,
        teamTwoScore: actions.teamTwoScore
      };
      const expected = {
        activeCards: actions.activeCards,
        teamOneScore: actions.teamOneScore,
        teamTwoScore: actions.teamTwoScore
      };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with the correct params on resetStore', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = actions.resetStore();
      mappedProps.resetStore();
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
      expect();
    });
  });
});
