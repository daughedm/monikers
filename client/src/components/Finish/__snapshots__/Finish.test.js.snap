// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`finish matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Finish
    history={
        Object {
            "push": [Function],
          }
    }
    resetStore={[Function]}
    teamNames={
        Array [
            "team1",
            "team2",
          ]
    }
    teamOneScore={1}
    teamTwoScore={2}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="finish-transition"
>
          <h2
                    className="win-headline"
                    style={
                              Object {
                                        "color": "#866AAD",
                                      }
                    }
          >
                    team2
          </h2>
          <h2
                    className="win-headline-two"
          >
                    WINS!
          </h2>
          <div
                    className="dashed-line"
          />
          <button
                    className="start-newgame-button"
                    onClick={[Function]}
                    type="submit"
          >
                    NEW GAME
          </button>
</div>,
        <div
          className="animation-one"
/>,
        <div
          className="animation-two"
/>,
      ],
      "className": "background-monikers",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="win-headline"
              style={
                            Object {
                                          "color": "#866AAD",
                                        }
              }
>
              team2
</h2>,
            <h2
              className="win-headline-two"
>
              WINS!
</h2>,
            <div
              className="dashed-line"
/>,
            <button
              className="start-newgame-button"
              onClick={[Function]}
              type="submit"
>
              NEW GAME
</button>,
          ],
          "className": "finish-transition",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "team2",
              "className": "win-headline",
              "style": Object {
                "color": "#866AAD",
              },
            },
            "ref": null,
            "rendered": "team2",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "WINS!",
              "className": "win-headline-two",
            },
            "ref": null,
            "rendered": "WINS!",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "dashed-line",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "NEW GAME",
              "className": "start-newgame-button",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "NEW GAME",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "animation-one",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "animation-two",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="finish-transition"
>
            <h2
                        className="win-headline"
                        style={
                                    Object {
                                                "color": "#866AAD",
                                              }
                        }
            >
                        team2
            </h2>
            <h2
                        className="win-headline-two"
            >
                        WINS!
            </h2>
            <div
                        className="dashed-line"
            />
            <button
                        className="start-newgame-button"
                        onClick={[Function]}
                        type="submit"
            >
                        NEW GAME
            </button>
</div>,
          <div
            className="animation-one"
/>,
          <div
            className="animation-two"
/>,
        ],
        "className": "background-monikers",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="win-headline"
                style={
                                Object {
                                                "color": "#866AAD",
                                              }
                }
>
                team2
</h2>,
              <h2
                className="win-headline-two"
>
                WINS!
</h2>,
              <div
                className="dashed-line"
/>,
              <button
                className="start-newgame-button"
                onClick={[Function]}
                type="submit"
>
                NEW GAME
</button>,
            ],
            "className": "finish-transition",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "team2",
                "className": "win-headline",
                "style": Object {
                  "color": "#866AAD",
                },
              },
              "ref": null,
              "rendered": "team2",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "WINS!",
                "className": "win-headline-two",
              },
              "ref": null,
              "rendered": "WINS!",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "dashed-line",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "NEW GAME",
                "className": "start-newgame-button",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "NEW GAME",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "animation-one",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "animation-two",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`finish matches the snapshot if teamOneScore is higher 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Finish
    history={
        Object {
            "push": [Function],
          }
    }
    resetStore={[Function]}
    teamNames={
        Array [
            "team1",
            "team2",
          ]
    }
    teamOneScore={3}
    teamTwoScore={2}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="finish-transition"
>
          <h2
                    className="win-headline"
                    style={
                              Object {
                                        "color": "#866AAD",
                                      }
                    }
          >
                    team1
          </h2>
          <h2
                    className="win-headline-two"
          >
                    WINS!
          </h2>
          <div
                    className="dashed-line"
          />
          <button
                    className="start-newgame-button"
                    onClick={[Function]}
                    type="submit"
          >
                    NEW GAME
          </button>
</div>,
        <div
          className="animation-one"
/>,
        <div
          className="animation-two"
/>,
      ],
      "className": "background-monikers",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="win-headline"
              style={
                            Object {
                                          "color": "#866AAD",
                                        }
              }
>
              team1
</h2>,
            <h2
              className="win-headline-two"
>
              WINS!
</h2>,
            <div
              className="dashed-line"
/>,
            <button
              className="start-newgame-button"
              onClick={[Function]}
              type="submit"
>
              NEW GAME
</button>,
          ],
          "className": "finish-transition",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "team1",
              "className": "win-headline",
              "style": Object {
                "color": "#866AAD",
              },
            },
            "ref": null,
            "rendered": "team1",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "WINS!",
              "className": "win-headline-two",
            },
            "ref": null,
            "rendered": "WINS!",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "dashed-line",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "NEW GAME",
              "className": "start-newgame-button",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "NEW GAME",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "animation-one",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "animation-two",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="finish-transition"
>
            <h2
                        className="win-headline"
                        style={
                                    Object {
                                                "color": "#866AAD",
                                              }
                        }
            >
                        team1
            </h2>
            <h2
                        className="win-headline-two"
            >
                        WINS!
            </h2>
            <div
                        className="dashed-line"
            />
            <button
                        className="start-newgame-button"
                        onClick={[Function]}
                        type="submit"
            >
                        NEW GAME
            </button>
</div>,
          <div
            className="animation-one"
/>,
          <div
            className="animation-two"
/>,
        ],
        "className": "background-monikers",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="win-headline"
                style={
                                Object {
                                                "color": "#866AAD",
                                              }
                }
>
                team1
</h2>,
              <h2
                className="win-headline-two"
>
                WINS!
</h2>,
              <div
                className="dashed-line"
/>,
              <button
                className="start-newgame-button"
                onClick={[Function]}
                type="submit"
>
                NEW GAME
</button>,
            ],
            "className": "finish-transition",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "team1",
                "className": "win-headline",
                "style": Object {
                  "color": "#866AAD",
                },
              },
              "ref": null,
              "rendered": "team1",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "WINS!",
                "className": "win-headline-two",
              },
              "ref": null,
              "rendered": "WINS!",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "dashed-line",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "NEW GAME",
                "className": "start-newgame-button",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "NEW GAME",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "animation-one",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "animation-two",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
